File: tests/fixtures/fixture_tauri/src/services/order.ts
import { invoke } from '@tauri-apps/api/core';

export interface Order {
  id: string;
  user_id: string;
  total: number;
}

export interface CreateOrderRequest {
  user_id: string;
  items: number[];
}

export async function placeOrder(userId: string, items: number[]): Promise<Order> {
  return await invoke('place_order', { userId, items });
}

export async function cancelOrder(orderId: string): Promise<boolean> {
  return await invoke('cancel_order', { orderId });
}

File: tests/fixtures/fixture_tauri/src-tauri/src/commands/order.rs
use crate::models::order::{Order, CreateOrderRequest, create_order, calculate_total};
use crate::utils::cache::{cache_get, cache_set};
use tauri::command;

#[command]
pub fn place_order(user_id: String, items: Vec<f64>) -> Result<Order, String> {
    let req = CreateOrderRequest { user_id, items };
    let order = create_order(req);
    cache_set(&format!("order:{}", order.id), &order.id);
    Ok(order)
}

#[command]
pub fn cancel_order(order_id: String) -> bool {
    let key = format!("order:{}", order_id);
    cache_get(&key).is_some()
}
